### PR TITLE
HCaptcha integration

### PR DESCRIPTION
**Pull request by nzambello**
Adds [HCaptcha](https://www.hcaptcha.com) integration.
HCaptcha is ligher than ReCaptcha and respects users' privacy, so it is a good alternative I'd like to use: now it's possible here.

Updated README with docs on how to use captchas.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -33,6 +33,11 @@ For each field, you can select the field type from:
 For every field you can set a label and a help text.
 For select, radio and checkbox fields, you can select a list of values.
 
+## Captcha verification
+
+This form addon is configured to work with [HCaptcha](https://www.hcaptcha.com) and [ReCaptch](https://www.google.com/recaptcha/) to prevent spam.
```
Comment (pnicolli) [inline README.md:38]: Typo: missing "a" at the end of "ReCaptch"

File: src/components/FormView.jsx [hunk 2]
```diff
@@ -1,5 +1,6 @@
 import React, { useCallback, useState, useEffect } from 'react';
 import { useIntl, defineMessages } from 'react-intl';
+import HCaptcha from '@hcaptcha/react-hcaptcha';
 import { GoogleReCaptcha } from 'react-google-recaptcha-v3';
```
Comment (pnicolli) [inline src/components/FormView.jsx:4]: How about a dynamic import based on the provided env vars?

File: src/components/FormView.jsx [hunk 3]
```diff
@@ -46,19 +47,20 @@ const FormView = ({
 }) => {
   const intl = useIntl();
 
-  const [loadedRecaptcha, setLoadedRecaptcha] = useState(null);
+  const [loadedCaptcha, setLoadedCaptcha] = useState(null);
   let validToken = '';
   const onVerifyCaptcha = useCallback(
     (token) => {
       // eslint-disable-next-line react-hooks/exhaustive-deps
+      console.log(token);
```
Comment (pnicolli) [inline src/components/FormView.jsx:55]: Imho, it would be better to remove these two lines

File: src/components/FormView.jsx [hunk 4]
```diff
@@ -148,6 +150,18 @@ const FormView = ({
                   </Grid.Row>
                 )}
 
+                {process.env.RAZZLE_HCAPTCHA_KEY && (
+                  <Grid.Row>
+                    <Grid.Column>
+                      <HCaptcha
+                        sitekey={process.env.RAZZLE_HCAPTCHA_KEY}
+                        onVerify={onVerifyCaptcha}
+                        size="invisible"
```
Comment (pnicolli) [inline src/components/FormView.jsx:159]: This should probably be an option for whoever uses this addon. The invisible captcha has privacy policy implications: [see hcaptcha suggestions here](https://docs.hcaptcha.com/faq#do-i-need-to-display-anything-on-the-page-when-using-hcaptcha-in-invisible-mode)

File: src/components/FormView.jsx (hunk 2 above)
Comment (tiberiuichim) [reply src/components/FormView.jsx:4]: I think wrapping in loadable() would be just fine.

Comment (pnicolli) [reply src/components/FormView.jsx:4]: Yeah that's right actually

File: README.md (hunk 1 above)
Comment (nzambello) [reply README.md:38]: Fixed in `main`

**Comment by nzambello** [comment]
For comments, see https://github.com/collective/volto-form-block/pull/8

**Pull request merged**